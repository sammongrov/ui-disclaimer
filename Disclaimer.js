import React from 'react';
import PropTypes from 'prop-types';

const Disclaimer = ({ Layout, content, onAccept, safeBgColors }) => (
  <Layout content={content} safeBgColors={safeBgColors} onAccept={() => onAccept()} />
);

Disclaimer.propTypes = {
  Layout: PropTypes.func.isRequired,
  content: PropTypes.string.isRequired,
  onAccept: PropTypes.func.isRequired,
  safeBgColors: PropTypes.array,
};

Disclaimer.defaultProps = {
  safeBgColors: [],
};

export default Disclaimer;
