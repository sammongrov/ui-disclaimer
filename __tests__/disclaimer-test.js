import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Text } from 'react-native';

import Disclaimer from '../Disclaimer';

const onAccept = jest.fn();
const Layout = ({ content, safeBgColors, onAccept }) => <Text onPress={onAccept}>{content}</Text>;
const content = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit...';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Disclaimer renders correctly', () => {
  beforeEach(() => {
    onAccept.mockClear();
  });

  it('with disclaimer.accepted === false', () => {
    const initialState = {
      disclaimer: {
        accepted: false,
      },
    };
    const store = mockStore(initialState);
    const tree = renderer
      .create(<Disclaimer Layout={Layout} content={content} onAccept={onAccept} store={store} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(onAccept.mock.calls.length).toBe(0);
  });

  it('with disclaimer.accepted === true', () => {
    const initialState = {
      disclaimer: {
        accepted: true,
      },
    };
    const store = mockStore(initialState);
    const tree = renderer
      .create(<Disclaimer Layout={Layout} content={content} onAccept={onAccept} store={store} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(onAccept.mock.calls.length).toBe(1);
    expect(onAccept.mock.calls[0][0]).toBeUndefined();
  });

  it('with disclaimer has accepted property', () => {
    const initialState = {};
    const store = mockStore(initialState);
    const tree = renderer
      .create(<Disclaimer Layout={Layout} content={content} onAccept={onAccept} store={store} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
    expect(onAccept.mock.calls.length).toBe(0);
  });
});

it('Disclaimer dispatches onAcceptDisclaimer', async () => {
  onAccept.mockClear();
  const initialState = {
    disclaimer: {
      accepted: false,
    },
  };
  const store = mockStore(initialState);
  const component = renderer.create(
    <Disclaimer Layout={Layout} content={content} onAccept={onAccept} store={store} />,
  ).root;

  await component.findByType(Text).props.onPress();

  expect(store.getActions()).toMatchSnapshot();
  expect(onAccept.mock.calls.length).toBe(1);
  expect(onAccept.mock.calls[0][0]).toBeUndefined();
});
