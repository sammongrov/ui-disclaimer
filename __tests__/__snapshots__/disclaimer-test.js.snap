// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disclaimer dispatches onAcceptDisclaimer 1`] = `Array []`;

exports[`Disclaimer renders correctly with disclaimer has accepted property 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  onPress={[Function]}
>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit...
</Text>
`;

exports[`Disclaimer renders correctly with disclaimer.accepted === false 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  onPress={[Function]}
>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit...
</Text>
`;

exports[`Disclaimer renders correctly with disclaimer.accepted === true 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  onPress={[Function]}
>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit...
</Text>
`;
